More control over which tokens to hide

amhed opened the pull request:
### PR Checklist (check all)

* [X] Updated `CHANGELOG.md` to describe the included fixes and changes made
* [X] Included issue number on the description of the PR
* [X] Commented on the relevant issue thread about this PR

### Issue number

#242 Hide 💩 Coints

### Changelog

### Added
* Actions that allow user to hide assets they don't want to see on their wallet

### Changed
* No longer automatically removing dust tokens / shit coins
* Moved styles from account balances view into their own file, per [suggestion](https://github.com/balance-io/balance-manager/issues/238#issuecomment-391414680)

amhed committed:
Fix #242 - Hide :shit: coins

- Keep track of coins that should be hidden on the client side
- Moved elements to styled components

amhed commented on src/views/Account/AccountBalances.js:
```diff
@@ -1,225 +1,61 @@
-    const allLowMarketTokensHaveNoValue =
-      tokensWithNoMarketValue.length === tokensToggleDisplay.length;
+
+    const emptyBalanceDisplay = '———';
```
Will probably create a `consts` file on an upper directory. This constant is used in other places as well.

amhed commented on src/views/Account/AccountBalances.js:
```diff
@@ -236,98 +72,136 @@ class AccountBalances extends Component {
+      ...tokensWithNoMarketValue,
+    ];
+
+    //const hiddenTokens = [...dustTokens.map(t => t.address), ...this.props.hiddenAssets];
```
Still not sure if we should automatically mark tokens that have low/no value ourselves or just leave to the user. If we do it automatically we need to keep track of two lists. I think I'll go with this: 
- If the local storage doesn't have any hiddenAssets, create the array and put in all shitcoins with low value
- After the local storage has an array just leave it up to the user

amhed commented:
Another thing we can do further down the line is integrate with http://tokentoilet.com (after its operational) so I can "flush" the tokens I don't want 🚽

ricburton commented:
That would be really cool.

Richard

Sent via Superhuman ( https://sprh.mn/?vip=richard@balance.io )

On Thu, May 31, 2018 at 22:40:23, Amhed Herrera < notifications@github.com > wrote:

> 
> 
> 
> Another thing we can do further down the line is integrate with http:/ / tokentoilet.
> com ( http://tokentoilet.com ) (after its operational) so I can "flush" the
> tokens I don't want 🚽
> 
> 
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub (
> https://github.com/balance-io/balance-manager/pull/254#issuecomment-393741328
> ) , or mute the thread (
> https://github.com/notifications/unsubscribe-auth/AAavkC5StDjNE478pOAiJF0p2qtQaZfeks5t4KmXgaJpZM4UQjLr
> ).
> 
> 
>

jinchung commented:
Hi @amhed thanks for your work. 
I've merged in your Spanish translations PR.

For this one, could we have a couple changes:
1. On default, we want to hide tokens with low market value. (Users can show them individually if they want).
2. We want the total balance to remain on the bottom instead of in between the shown and hidden tokens.

Otherwise it is looking good! 

Thanks!

Jin

amhed commented:
Thanks for feedback @jinchung. Will tackle this later today.

ricburton commented:
🙌 Preview this pull request.

Built with commit 581f00d8116c97fcd003579811d771739035d438

https://deploy-preview-254--balance-manager.netlify.com

amhed committed:
Always show total row at the bottom

amhed commented:
> 1. On default, we want to hide tokens with low market value. (Users can show them individually if they want).

What's the desired behavior here? if there's a low/zero market value token that the user decides to show, should that be persisted on local storage as well so that when I refresh my screen I see it again? or should it just be shown for the current session

> 2. Show totals row at the end instead: 

![image](https://user-images.githubusercontent.com/579291/41201945-20a8be7c-6c87-11e8-837b-9f0f46cee759.png)

ricburton commented:
Good points. I want to see my Queen Jin Token 👑
On 10 Jun 2018, 09:22 -0400, Amhed Herrera <notifications@github.com>, wrote:
> >
> > 1. On default, we want to hide tokens with low market value. (Users can show them individually if they want).
> >
> What's the desired behavior here? if there's a low/zero market value token that the user decides to show, should that be persisted on local storage as well so that when I refresh my screen I see it again? or should it just be shown for the current session
> >
> > 2. Show totals row at the end instead:
> >
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.

amhed committed:
Better management of shit tokens

Keeping two lists on local storage
  - Tokens I want to hide
  - Tokens I want to show, regardless if they have low market value or not

Sorting no value and low market value tokens

Adding separator logic

amhed commented:
Went ahead with two lists on local storage http://recordit.co/oa8erGh5lq

jinchung commented:
Hey @amhed ! Thanks a lot! Sorry for the delay - been swamped with our next launch.

@ricburton @christianbaroni  I notice that when I expand to see my hidden tokens, there are still ones with market value that are part of the hidden ones. Now, I know we had a set number of tokens to be visible, but I do wonder, if I expand everything, I do find it a bit odd that the ones with market values are still mixed in with the hidden ones. What do you guys think should be the right UI here? 

<img width="1057" alt="screen shot 2018-06-14 at 3 19 16 pm" src="https://user-images.githubusercontent.com/1285228/41436294-d1a1707c-6fe6-11e8-8ac8-ac50b0b3bdb2.png">

amhed commented:
I think we can: 
- Order the tokens. Show the ones with no market value first, followed by the ones with value
- Separate them with a thin line, so that it's visually easy to spot there's a difference

I can submit an example tonight

jinchung commented:
Thanks! Also, would you mind catching up with latest? There's a lot of new stuff that's in there now.

amhed commented:
@jinchung added logic to sort and separate. Need a bit of help with design, I'm terrible at UX :D

jinchung commented:
hey @amhed thanks a lot! sorry I was on vacation for a bit - will ask Christian for a design review as he'll be best for this.

jinchung commented:
@christianbaroni how do you feel about the placement of the "Hide tokens" and the blue line that delineates between tokens of value or not, and the default limited display of some of the tokens with value but not all of them?  And anything else :)

ricburton commented:
@amhed I am keen to pair with you on this when you come by the studio!

kwikiel commented:
<img width="1015" alt="screen shot 2018-07-04 at 23 08 53" src="https://user-images.githubusercontent.com/5274399/42294163-402180c0-7fdf-11e8-9913-9c12f4359f55.png">

I would like to hide some tokens with substantial market value (those Magic tokens are virtual - I cannot send or sell them and they are not trade-able anywhere)

I would like to create proper PR myself but I have no idea how this should be handled from architecture side. Hiding them and not adding those to totals? Storing such things in local storage? 

And doing at every time after browser purge? 

There may be some way of keeping "settings" but it's again hard from security and privacy perspective
